src/builtins/core/timezone.rs

```diff
@@ -140,24 +142,24 @@ impl TimeZone {
         instant: &Instant,
         provider: &impl TimeZoneProvider,
     ) -> TemporalResult<IsoDateTime> {
-        let nanos = self.get_offset_nanos_for(instant.as_i128(), provider)?;
-        IsoDateTime::from_epoch_nanos(instant.epoch_nanoseconds(), nanos.to_i64().unwrap_or(0))
+        let nanos = self.get_offset_nanos_for(instant.epoch_nanoseconds(), provider)?;
+        IsoDateTime::from_epoch_nanos(instant.epoch_nanoseconds(), nanos.as_i128().to_i64().unwrap_or(0))
```

Comment:
thought: I think `to_i64` could be implemented directly on `EpochNanoseconds` as a method. We're already doing the bound check, so it could be a nice addition from an ergonomics perspective.